### PR TITLE
Update OTG telemetry_interface_packet_counters_test with updates from the ATE version

### PR DESCRIPTION
**Pull request by jasdeep-hundal**

### CODE REVIEW

File: feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go [hunk 1]
```diff
@@ -106,53 +106,63 @@ func TestInterfaceCounters(t *testing.T) {
...
 		path    string
-		counter *ygnmi.Value[uint64]
+		counter ygnmi.SingletonQuery[uint64]
+		skip    bool
 	}{{
 		desc:    "InUnicastPkts",
 		path:    intfCounterPath + "in-unicast-pkts",
-		counter: gnmi.Lookup(t, dut, intfCounters.InUnicastPkts().State()),
+		counter: intfCounters.InUnicastPkts().State(),
+	}, {
```
Comment (thesrinath) [inline feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go:115]: why is this repeated twice?

File: feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go [hunk 2]
```diff
@@ -240,6 +250,18 @@ func TestIntfCounterUpdate(t *testing.T) {
 		// "ipv4":   subintf2.Ipv4().Counters().OutPkts().Get(t),
 		// "ipv6":   subintf2.Ipv6().Counters().OutPkts().Get(t),
 	}
+	if *deviations.InterfaceCountersFromContainer {
+		dutInPktsBeforeTraffic = map[string]uint64{
+			"parent": *gnmi.Get(t, dut, i1.Counters().State()).InUnicastPkts,
+			// "ipv4":   *subintf1.Ipv4().Counters().Get(t).InPkts,
```
Comment (thesrinath) [inline feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go:256]: remove lines 256,257 and 261,262 if not relevant

File: feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go [hunk 3]
```diff
@@ -315,6 +337,18 @@ func TestIntfCounterUpdate(t *testing.T) {
 		// "ipv4":   subintf2.Ipv4().Counters().OutPkts().Get(t),
 		// "ipv6":   subintf2.Ipv6().Counters().OutPkts().Get(t),
 	}
+	if *deviations.InterfaceCountersFromContainer {
+		dutInPktsAfterTraffic = map[string]uint64{
+			"parent": *gnmi.Get(t, dut, i1.Counters().State()).InUnicastPkts,
+			// "ipv4":   *subintf1.Ipv4().Counters().Get(t).InPkts,
```
Comment (thesrinath) [inline feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go:343]: ditto here and below

File: feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go (hunk 1 above)
Comment (jasdeep-hundal) [reply feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go:115]: Also present in the ATE version of the test, I've now deleted it here.

File: feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go (hunk 2 above)
Comment (jasdeep-hundal) [reply feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go:256]: These are associated with the TODO at line 238 (https://github.com/openconfig/featureprofiles/pull/1415/files#diff-9c42efab3f06ea8f988e62604a01384d33c1c2f79bc98a13d4dcf67543053a7cR238), which also applies to the ATE test that also has these lines commented out.

I'd err on the side of keeping the extra noise here to make reviews of updates that apply for both tests easier.

File: feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go (hunk 3 above)
Comment (jasdeep-hundal) [reply feature/gnmi/otg_tests/telemetry_interface_packet_counters_test/telemetry_interface_packet_counters_test.go:343]: See above comment.

**Comment by jasdeep-hundal** [comment]
@thesrinath : Responded. Added the deviation from https://github.com/openconfig/featureprofiles/pull/1407 as well, will test before merging.

**Review by thesrinath — APPROVED** [review]

**Pull request merged**